src/MongoDB.Driver/Encryption/AutoEncryptionLibMongoController.cs

```diff
@@ -223,5 +227,39 @@ private void RestoreDbNodeInResponse(BsonDocument request, BsonDocument response
                 }
             }
         }
+
+        private void WaitForMongocryptdReady()
```

Comment:
We still may have a reason to improve(fix?) SDAM logic. But this logic is probably good enough for now. At least to finally avoid timeout exception for FLE tests.